otter/grade/utils.py

```diff
@@ -19,6 +19,10 @@
 
 SCORES_DICT_PERCENT_CORRECT_KEY = "percent_correct"
 
+SCORES_DICT_SUMMARY_KEY = "summary"
+
+SCORES_DICT_GRADING_STATUS_KEY = "grading_status"
```

Comment:
Suggested future refactor (not as part of this PR): instead of using a dict with several hard-coded keys, we can use a dataclass to hold the scores and all of this metadata. If you've got time/are inclined, I'm happy to review a PR, otherwise I'll get to it at some point in the future.

(No changes required for this comment in this PR -- feel free to mark as resolved immediately)

Reply:
I have time this week I think to knock this out. I am assuming you mean a custom class for the scores and metadata. unless "dataclass" is a thing I don't know about?

Reply:
I mean using Pythons [dataclasses library](https://docs.python.org/3/library/dataclasses.html). ex: https://github.com/ucbds-infra/otter-grader/blob/master/otter/test_files/abstract_test.py#L12